### ISSUE TITLE
Negative character classes behave wrong when case insensitive

### ISSUE THREAD

**Issue by pvginkel** [rust-lang/regex#99]
In the following example, **both** the third and forth assertion both fail:

```rust
extern crate regex;

use self::regex::Regex;

fn main() {
    let rs = Regex::new("[^x]").ok().unwrap();
    assert!(!rs.is_match("x"));
    assert!(rs.is_match("X"));

    let rs = Regex::new("(?i)[^x]").ok().unwrap();
    assert!(!rs.is_match("x"));
    assert!(!rs.is_match("X"));
}
```

**Comment by BurntSushi** [comment]
@pvginkel Thanks for the report! This was surprisingly non-trivial to fix. See the [commit message for the horrifying details](https://github.com/rust-lang/regex/commit/56ea4a835c4a1e9b3193c74537a250d136bc127d).

**Comment by BurntSushi** [comment]
I've published the fix in [`regex-syntax` 0.1.3](https://crates.io/crates/regex-syntax).

**Comment by pvginkel** [comment]
Thank you for fixing this, and this fast.
